server/sonar-server/src/main/java/org/sonar/server/computation/qualitygate/ResolvedCondition.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.qualitygate;
+
+import com.google.common.base.Objects;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import org.sonar.server.computation.metric.Metric;
+
+import static java.util.Objects.hash;
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class ResolvedCondition {
+
+  private final long id;
+  private final Metric metric;
+  private final String operator;
+  @CheckForNull
+  private final String warningThreshold;
+  @CheckForNull
+  private final String errorThreshold;
+  @CheckForNull
+  private final Integer period;
+
+  public ResolvedCondition(long id, Metric metric, String operator,
+    @Nullable Integer period, @Nullable String errorThreshold, @Nullable String warningThreshold) {
+    this.id = id;
+    this.metric = requireNonNull(metric);
+    this.operator = requireNonNull(operator);
+    this.period = period;
+    this.errorThreshold = errorThreshold;
+    this.warningThreshold = warningThreshold;
+  }
+
+  public long getId() {
+    return id;
+  }
+
+  public Metric getMetric() {
+    return metric;
+  }
+
+  public String getOperator() {
+    return operator;
+  }
+
+  public String getWarningThreshold() {
```

Comment:
I realize annotations are missing on the getters of properties annotated with `@CheckForNull`, IDEA did not generate them